File: solutions/cpp/armstrong-numbers/1/armstrong_numbers.cpp
#include <cmath>
#include "armstrong_numbers.h"

namespace armstrong_numbers
{
    bool is_armstrong_number(int num)
    {
        if (num == 0) return true;

        int num_digits = std::log10(num) + 1;
        int sum = 0;

        for (int rem = num; rem != 0; rem /= 10)
        {
            sum += std::pow(rem % 10, num_digits);
        }

        return sum == num;
    }
}

File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        // Convert number to string to easily get each digit and count the digits
        String numStr = Integer.toString(numberToCheck);
        int numDigits = numStr.length();
        int sum = 0;
        
        // Iterate through each digit
        for (char c : numStr.toCharArray()) {
            int digit = Character.getNumericValue(c);
            sum += Math.pow(digit, numDigits);  // Raise the digit to the power of the number of digits
        }

        // If the sum of the powered digits is equal to the original number, return true
        return sum == numberToCheck;
    }

}
